source/views/calendar/times.js

```diff
@@ -41,3 +41,45 @@ export function times(event: EventType) {
 
   return {start, end, allDay}
 }
+
+export function detailTimes(event: EventType) {
+  const eventLength = moment
+    .duration(event.endTime.diff(event.startTime))
+    .asHours()
+
+  const allDay = eventLength === 24
+  const multiDay = event.startTime.dayOfYear() !== event.endTime.dayOfYear()
+  const sillyZeroLength = event.startTime.isSame(event.endTime, 'minute')
+  const endsOnSameDay = event.startTime.isSame(event.endTime, 'day')
+
+  const endFormat = endsOnSameDay ? 'h:mm A' : 'MMM. D h:mm A'
+  let startTimeFormatted = event.startTime.format('MMM. D h:mm A')
+  let endTimeFormatted = event.endTime.format(endFormat)
+  let midnightTime = '12:00 AM'
+
+  let start, end
+  if (event.isOngoing) {
+    start = event.startTime.format('MMM. D')
+    end = event.endTime.format('MMM. D')
+  } else if (multiDay) {
+    // 12:00 PM to Jun. 25 3:00pm
+    // Midnight to Jun. 25 <-- assuming the end time is also midnight
+    start = startTimeFormatted
+    const endFormat = endTimeFormatted === midnightTime
+      ? 'MMM. D'
+      : 'MMM. D h:mm A'
+    end = `${event.endTime.format(endFormat)}`
+  } else if (sillyZeroLength) {
+    start = `Starts at ${startTimeFormatted}`
```

Comment:
`startTimeFormatted` will have a date in this case.

I'm going to suggest changing the word "on" for "at".
> **Current** "Starts `at` Aug. 13 12:00 PM"
> **Proposed**: "Starts `on` Aug. 13 12:00 PM"